src/datadog/config_manager.cpp

```diff
@@ -215,7 +313,19 @@ std::vector<ConfigMetadata> ConfigManager::apply_update(
     rule.rate = *rate;
     rule.matcher = catch_all;
     rule.mechanism = SamplingMechanism::RULE;
-    rules.emplace(rules.cbegin(), std::move(rule));
+
+    // Convention: Catch-all rules should ALWAYS be the last in the list.
+    // If a catch-all rule already exists, replace it.
+    // If NOT, add the new one at the end of the rules list.
+    if (rules.empty()) {
+      rules.emplace_back(std::move(rule));
+    } else {
+      if (auto& last_rule = rules.back(); last_rule.matcher == catch_all) {
+        last_rule = rule;
```

Comment:
Am I understanding correctly that you've passed `rules.back()` by reference, so after this line `rules.back()` now points to the new TraceSamplerRule you constructed?